prboom2/src/d_main.c

```diff
@@ -1483,17 +1483,15 @@ static void D_AutoloadIWadDir()
 
 static void D_AutoloadPWadDir()
 {
-  int p = M_CheckParm("-file");
-  if (p)
-  {
-    while (++p != myargc && myargv[p][0] != '-')
+  int i;
+  for (i = 0; i < numwadfiles; ++i)
+    if (wadfiles[i].src == source_pwad)
```

Comment:
We should probably change the second argument to `D_AddFile()` in `AutoLoadWADs()` to `source_auto_load`, because else I am afraid we end up auto-loading WADs for auto-loaded WADs.